src/main/java/com/hierynomus/smbj/share/Directory.java

```diff
@@ -160,10 +163,69 @@ private F prepareNext() {
                 );
             }
         }
-
         @Override
         public void remove() {
             throw new UnsupportedOperationException();
         }
     }
+
+    public void rename(String newName)throws TransportException, SMBApiException {
+        this.rename(newName, false);
+    }
+
+    public void rename(String newName, boolean replaceIfExist)throws TransportException, SMBApiException {
+        this.rename(newName, replaceIfExist, 0);
+    }
+
+    public void rename(String newName, boolean replaceIfExist, long rootDirectory)throws TransportException, SMBApiException {
+        FileRenameInformation renameInfo = new FileRenameInformation(replaceIfExist, rootDirectory, newName);
+        this.setFileInformation(renameInfo);
+    }
+
+    /**
+     * Get information for a given fileId
+     **/
+    private <F extends FileSettableInformation> void setFileInformation(F information) throws SMBApiException, TransportException {
```

Comment:
Extract to package private method on DiskEntry (the common superclass of File and Directory), also for `setInfoCommon`